readme.md

```diff
@@ -59,6 +59,24 @@ and executes logic from the `protocols` package. Data required for the secure cr
 More detailed information can be found in each package's respective _readme_.
 ![architecture](./client/architecture.png)
 
+## On-chain code
+
+`go-nitro` was originally being developed "off-chain first". Alongside that development, we are now building out the on-chain component (solidity contracts): these are housed in the `nitro-protocol` directory. This directory contains an npm package with a hardhat / typechain / jest toolchain. This code originates from the nitro-protocol [npm package](https://www.npmjs.com/package/@statechannels/nitro-protocol), but is being modified / upgraded. The new version of the code is sometimes called nitro v2 or "satp".
+
+Initially, these two development streams will remain independent, with each having its own continuous integration checks.
+
+Over time, the development streams will become more coupled.
+
+We now use `solc` and `abigen` to generate Go bindings from our `*.sol` source files. This is achieved by running the `generate_adjudicator_bindings.sh` script at the top level of the repo. Because our `*.sol` files depend on external projects via `node_modules`, to run this script successfully you must:
+
+- have successfully run `npm install` in the `nitro-protocol` directory.
+- have [solc](https://docs.soliditylang.org/en/v0.8.13/installing-solidity.html) installed at the correct version (currently 0.7.4, see the CI or linting config for a hint if you think it may have changed)
```

Comment:
On my machine, `solc` is installed via homebrew. I am only able to install solidity 0.7.6 with homebrew as that is the latest 0.7 version.  Is it worth it to bump the contracts to the latest 0.7 version?

Reply:
You can use [`solc-select`](https://github.com/crytic/solc-select) to install multiple specific versions of `solc`.

Reply:
One of the recommendations from our audit was to pin the solidity compiler version (so that we get a deterministic deployment from the audited source code). So I think it would be better for folks to be able to install a specific version of `solc`. There are instructions [here](https://docs.soliditylang.org/en/v0.8.13/installing-solidity.html#macos-package) for doing that with homebrew.

Reply:
Unfortunately, I haven't succeeded in installing Solidity version 0.7.4 from Homebrew. I am seeing an install error.

In general, I agree with pinning to a specific solidity version. For this specific case, I would be in favor of pinning at solidity 0.7.6 as the minor 0.7 version seems to have settled on patch version 0.7.6 a year and a half ago.

Reply:
We are planning on upgrading to `0.8` soon #556 .